crates/flux-infer/src/infer.rs

```diff
@@ -224,7 +231,7 @@ impl<'genv, 'tcx> InferCtxtRoot<'genv, 'tcx> {
             flux_config::SmtSolver::CVC5 => liquid_fixpoint::SmtSolver::CVC5,
         };
 
-        fcx.check(cache, cstr, self.opts.scrape_quals, backend)
+        fcx.check(cache, cstr, kind, self.opts.scrape_quals, backend)
```

Comment:
We could also use the `kind` to generate the name for the constraint file. I think we are overriding it right now.